src/main/msp/msp.c

```diff
@@ -1017,6 +1017,17 @@ static bool mspProcessOutCommand(uint8_t cmdMSP, sbuf_t *dst)
         }
         break;
 
+    case MSP_MODE_RANGES_EXTRA:
+        sbufWriteU8(dst, 3);    // prepend number of extra bytes per available 'slot'
```

Comment:
Hehe, looks like there was a misunderstanding. The idea is to send the length of the following array (i.e. `MAX_MODE_ACTIVATION_CONDITION_COUNT`), and not the length of the structure.
The length of the structure is defined by the MSP format and will not be changed, so knowing the length of the array will allow a client to read all of it and know when to stop, so it is possible to append more data after the end of the array in a later MSP version.

Reply:
MSP length is known, sending length on entry allows future extension (on entry basis)...

Reply:
ok. I can see how that makes it more readable and more immune to error with future expansion.